app/models/patient.rb

```diff
@@ -32,10 +32,16 @@ class Patient
   mongoid_userstamp user_model: 'User'
 
   # Methods
+
+
   def self.search(name_or_phone_str) # TODO: optimize
-    name_matches = Patient.where name: name_or_phone_str
-    primary_matches = Patient.where primary_phone: name_or_phone_str
-    secondary_matches = Patient.where secondary_phone: name_or_phone_str
-    (name_matches | primary_matches | secondary_matches)
+    clean_phone = name_or_phone_str.gsub(/\D/, '')
+    phone = clean_phone[0..2] + '-' + clean_phone[3..5] + '-' + clean_phone[6..10]
+    begin
+      name_matches = Patient.where ({name: /^#{Regexp.escape(name_or_phone_str)}$/i})
```

Comment:
(And I see that you beat me to the thing I posted about below! Long-winded explanation of the problem and how to solve it is below, let me know if you have any questions or if I can unstick anything.)